### PR TITLE
🌱 Replace Sigstore library calls with REST API

### PR DESCRIPTION
**Pull request by azeemshaikh38**
This PR replaces any Cosign/Rekor library calls with Rekor's public API calls. Using the library is not fully supported and is causing failures in e2e tests. Also, the amount of transitive dependencies added when using Sigstore as a library, makes it worth the effort to copy/paste some basic functions instead of importing them.

### CODE REVIEW

File: app/post_results.go [hunk 1]
```diff
@@ -259,13 +255,85 @@ func extractAndVerifyCertForPayload(ctx context.Context,
...
+		Hash: fmt.Sprintf("sha256:%s", hex.EncodeToString(payloadSHA[:])),
+	}
+	jsonPayload, err := json.Marshal(rekorPayload)
+	if err != nil {
+		return nil, fmt.Errorf("marshaling json payload: %w", err)
+	}
+
+	rekorReq, err := http.NewRequestWithContext(ctx,
+		http.MethodPost,
+		"https://rekor.sigstore.dev/api/v1/index/retrieve",
```
Comment (naveensrinivasan) [inline app/post_results.go:286]: We are using `prod`. Maybe use the `staging`  https://rekor.sigstage.dev.  Here is something that is being used https://github.com/slsa-framework/slsa-github-generator/pull/433

File: app/post_results.go [hunk 2]
```diff
@@ -259,13 +255,85 @@ func extractAndVerifyCertForPayload(ctx context.Context,
...
+		return nil, fmt.Errorf("decoding Rekor response: %w", err)
+	}
+
+	return rekorResult, nil
+}
+
+func getTLogEntry(ctx context.Context, uuid string) (*tlogEntry, error) {
+	rekorReq, err := http.NewRequestWithContext(ctx,
+		http.MethodGet,
+		fmt.Sprintf("https://rekor.sigstore.dev/api/v1/log/entries/%s", uuid),
```
Comment (naveensrinivasan) [inline app/post_results.go:310]: The same comment as above. We should probably use the `staging`

File: app/post_results.go (hunk 1 above)
Comment (azeemshaikh38) [reply app/post_results.go:286]: Why would we use `staging`? Isn't `prod` the stable version?

Comment (naveensrinivasan) [reply app/post_results.go:286]: @laurentsimon/ @asraa can add more info on this. It was recommended that we use staging as prod is having some issues.

File: app/post_results.go (hunk 2 above)
Comment (bobcallaway) [reply app/post_results.go:310]: I disagree, staging can get wiped and may be unstable; `rekor.sigstore.dev` is meant to be the production endpoint (albeit we are still operating in a beta capacity).

Comment (naveensrinivasan) [reply app/post_results.go:310]: OK, Thanks

**Pull request merged**